public/docs/ts/latest/guide/component-styles.jade

```diff
@@ -294,54 +304,61 @@ code-example(format='').
-  We move file folders around all the time during the evolution of our applications. 
-  It's no fun patching the style and template URLs when we do.
-  
-  ### *moduleId*
-  
-  We can change the way Angular calculates the full URL be setting the component metadata's `moduleId` property.
-  
-  If we knew the component file's base path, we'd set `moduleId` to that and 
-  let Angular construct the full URL from this base path plus the CSS and template file names.
-  
-  Our challenge is to calculate the base path with minimal effort. 
-  If it's too hard, we shouldn't bother; we should just write the full path to the root and move on.
+  +makeExample('component-styles/ts/app/quest-summary.component.ts', 'urls')(format='.')
 
-  Fortunately, *certain* module loaders make it easy.
-  SystemJS (starting in v.0.19.19) sets a special `__moduleName` variable to the URL of the component file.
-.alert.is-important
   :marked
-    Caution: we currently regard the *__moduleName* feature as experimental.
-:marked
-  Now it's trivial to set the `moduleId` to the `__moduleName` and write module-relative paths for style and template URLs.
-  
-+makeExample('component-styles/ts/app/quest-summary.component.ts','', 'app/quest-summary.component.ts')
+    We can't do that by default. Angular can't find the files and throws an error:
+
+    `EXCEPTION: Failed to load quest-summary.component.html`
+
+    Why can't Angular calculate the HTML and CSS URLs from the component file's location? 
+
+    Unfortunately, that location is not readily known. 
+    Angular apps can be loaded in many ways: from individual files, from SystemJS packages, or
+    from CommonJS packages, to name a few. 
+    With this diversity of load strategies, it's not easy to tell at runtime where these files actually reside.
+
+    The only location Angular can be sure of is the URL of the `index.html` home page.
+    So by default it resolves template and style paths relative to the URL of `index.html`.
+    That's why we previously wrote our CSS file URLs with an `app/` base path prefix.
+
+    Although this works with any code loading scheme, it is very inconvenient.
+    We move file folders around all the time during the evolution of our applications. 
+    It's no fun patching the style and template URLs when we do.
+
+    ### *moduleId*
+
+    We can change the way Angular calculates the full URL be setting the component metadata's `moduleId` property.
+
+    If we knew the component file's base path, we'd set `moduleId` to that and 
+    let Angular construct the full URL from this base path plus the CSS and template file names.
+
+    Our challenge is to calculate the base path with minimal effort. 
+    If it's too hard, we shouldn't bother; we should just write the full path to the root and move on.
+
+    Fortunately, *certain* module loaders make it easy.
+    SystemJS (starting in v.0.19.19) sets a special `__moduleName` variable to the URL of the component file.
```

Comment:
I think we can kill this whole part of the text, in Dart you can specify paths relative to your component  folder. The Angular transformer will handle everything for you.